pkg/util/resources.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+)
+
+// ParseResourcesString takes a comma separated string list of resources in the form of
+// <api-group1>, <api-group2> .. and returns a map[string]bool
+func ParseResourceGroupsString(resourceGroups string) map[string]bool {
+	if resourceGroups == "" {
+		return nil
+	}
+
+	// trim single and double quotes to make this safe from
+	// user passing the resource option quoted in the container args
+	resourceGroups = strings.Trim(resourceGroups, `'"`)
+
+	groupsMap := map[string]bool{}
+	for _, g := range strings.Split(resourceGroups, ",") {
+		groupsMap[strings.Trim(g, " ")] = true
```

Comment:
I am also not clear on why spaces would need to be trimmed here.

Reply:
removed the trimming